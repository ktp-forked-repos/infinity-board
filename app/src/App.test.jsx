import ReactDOM from 'react-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import Board from './board/board';
import DraggableBox from './board/draggable-box';

/**
 * @jest-environment node
*/
/* eslint-disable */
it('renders correctly', () => {
  const tree = renderer.create(<Board />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('generates boxes', () => {
  const doc = renderer.create(<Board />);
  const instance = doc.root.instance;
  const mockEvent = {
    target: { dataset: { type: 'text' } },
    clientX: 50,
    clientY: 50,
  };

  instance.generateBox(mockEvent);
  expect(doc.toJSON()).toMatchSnapshot();
});
