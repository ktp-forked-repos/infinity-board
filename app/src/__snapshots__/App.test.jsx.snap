// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates boxes 1`] = `
<div>
  <button
    data-type="text"
    onClick={[Function]}
  >
    Box
  </button>
  <button
    data-type="image"
    onClick={[Function]}
  >
    Image
  </button>
  <div>
    <div
      className="Box"
      onBlur={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "backgroundColor": "#ff000",
          "cursor": "default",
          "height": "160px",
          "left": "50px",
          "padding": "20px",
          "top": "50px",
          "visibility": "visible",
          "width": "160px",
          "zIndex": 1,
        }
      }
    >
      <div
        className="Text-box unselectable"
      >
        
      </div>
    </div>
  </div>
  // 
  <div
    className="Wrapper"
    data-type="image"
    onDrop={[Function]}
    style={
      Object {
        "visibility": "visible",
        "zIndex": 2,
      }
    }
  />
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <button
    data-type="text"
    onClick={[Function]}
  >
    Box
  </button>
  <button
    data-type="image"
    onClick={[Function]}
  >
    Image
  </button>
  // 
  <div
    className="Wrapper"
    data-type="image"
    onDrop={[Function]}
    style={
      Object {
        "visibility": "visible",
        "zIndex": 2,
      }
    }
  />
</div>
`;
